### PR TITLE
Removed all the duplicate icon rules

### PR DESCRIPTION
**Pull request by jimmynotjim**
The latest changes to cf-icons made the duplicate icon rules for links and buttons unnecessary.

## Removals

- Duplicate icon code from cf-buttons
- Duplicate icon code from cf-typography

## Testing

- checkout this branch, be sure you've npm linked these modules by running `npm run cf-link`
- checkout the `example/remove-icons` branch within the sandbox
- in that repo run `npm run cf-link && npm run build && npm start`
- navigate to `http://localhost:3000/components/cf-buttons/` and check the __Icon Buttons__ section
- navigate to `http://localhost:3000/components/cf-typography/` and check out the __Links__ section

## Review

- @Scotchester 

## Screenshots

Everything should look identical to before.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/cf-typography/src/atoms/links.less [hunk 1]
```diff
@@ -109,8 +39,6 @@
 
         position: relative;
 
-        padding-right: 1.25em;
-
         &:after,
         &:before {
```
Comment (Scotchester) [inline src/cf-typography/src/atoms/links.less]: With the new way icons are being handled, the `&:before` selector can be removed from this group.

File: src/cf-typography/src/atoms/links.less [hunk 2]
```diff
@@ -126,16 +54,17 @@
             left: 0;
         }
```
Comment (Scotchester) [inline src/cf-typography/src/atoms/links.less]: With the new way icons are being handled, this whole selector can be removed.

File: src/cf-typography/src/atoms/links.less [hunk 3]
```diff
@@ -126,16 +54,17 @@
...
 
-        &.a-link__before {
-            padding-right: 0;
+        &.cf-icon__after {
+            padding-right: 1.25em;
+        }
+
+        &.cf-icon__before {
             padding-left: 1.25em;
         }
```
Comment (Scotchester) [inline src/cf-typography/src/atoms/links.less]: With the new way icons are being handled, this whole selector can be removed.

File: src/cf-typography/src/atoms/links.less [hunk 4]
```diff
@@ -147,7 +76,12 @@
...
             }
 
-            &.a-link__before {
+            &.cf-icon__after {
+                padding-left: 1em;
+                padding-right: 2.25em;
+            }
+
+            &.cf-icon__before {
                 padding-left: 2.25em;
```
Comment (Scotchester) [inline src/cf-typography/src/atoms/links.less]: With the new way icons are being handled, this line and the `padding-left: 1em;` in `&.cf-icon__after` can be deleted and the `padding-left: 1em` can be restored to the parent `&.a-link__bg` selector.

File: src/cf-typography/src/atoms/links.less (hunk 1 above)
Comment (jimmynotjim) [inline src/cf-typography/src/atoms/links.less]: Are you sure? This is for positioning the icon when it's in a jump link at small sizes, which is essentially a block link situation.

File: src/cf-typography/src/atoms/links.less (hunk 2 above)
Comment (jimmynotjim) [inline src/cf-typography/src/atoms/links.less]: Same comment as above, this is necessary for the icon to be placed appropriately inside a block link

Comment (Scotchester) [inline src/cf-typography/src/atoms/links.less]: Yeah, if you take the recommendations in all of my comments, it works. I tested it out in the DevTools.

**Comment by Scotchester** [comment]
Paired and realized that my comments did not take into account links that wrap onto multiple lines. All comments retracted!

File: src/cf-buttons/src/atoms/buttons-with-icons.less [hunk 5]
```diff
@@ -1,28 +1,22 @@
...
+.a-btn_icon-position-left:before {
     padding-right: unit( 10.5px / @btn-font-size, em );
     border-right: 1px solid mix( @btn-bg, @btn-text, 50% );
     margin-right: unit( 7px / @btn-font-size, em );
-  }
 }
 
-.a-btn_icon-position-right {
-  &:after  {
+.a-btn_icon-position-right:after  {
```
Comment (Scotchester) [inline src/cf-buttons/src/atoms/buttons-with-icons.less]: Discussed while pairing, but for the record: renaming these to be a modifier, now that modifiers are not needed at the choose-an-icon level (see removed code below).

We settled on `.a-btn__icon-on-left` and `.a-btn__icon-on-right`.

**Comment by jimmynotjim** [comment]
@Scotchester I think I've covered everything here and discussed during our call.

File: src/cf-buttons/src/atoms/buttons-with-icons.less [hunk 6]
```diff
@@ -2,31 +2,31 @@
...
+    }
 
-  .a-btn__disabled&,
-  .a-btn[disabled]& {
-    border-color: mix( @btn__disabled-bg, @btn__disabled-text, 50% );
-  }
+    .a-btn__disabled&,
+    .a-btn[disabled]& {
+        border-color: mix( @btn__disabled-bg, @btn__disabled-text, 50% );
+    }
```
Comment (Scotchester) [inline src/cf-buttons/src/atoms/buttons-with-icons.less]: Hooray for stealth indentation fixes! 😀

File: src/cf-buttons/usage.md [hunk 7]
```diff
@@ -348,77 +348,152 @@ For accessibility reasons, use the semantic `<button>` instead of a link when po
 
 #### Button with icon on the left
 
-<button class="a-btn a-btn_icon__left cf-icon cf-icon-delete">
-    Close
+<button class="cf-icon
+               cf-icon__before
+               cf-icon-delete
+               a-btn
+               a-btn__icon-on-left">
```
Comment (Scotchester) [inline src/cf-buttons/usage.md]: Hrmmmmmmmm I kinda feel like the button classes should come first, since that's what the thing _is_. Thoughts?

Comment (jimmynotjim) [inline src/cf-buttons/usage.md]: No real opinion but looking at it, it makes sense since we're defining buttons here and the icon is a modifier.

Comment (Scotchester) [inline src/cf-buttons/usage.md]: Cool. Make that change here and in the icon links, and then we're good to :shipit:

Comment (jimmynotjim) [inline src/cf-buttons/usage.md]: Done

**Comment by Scotchester** [comment]
:+1: :shipit:

**Pull request merged**